django/utils/html.py

```diff
@@ -358,3 +358,19 @@ def avoid_wrapping(value):
     spaces where there previously were normal spaces.
     """
     return value.replace(" ", "\xa0")
+
+
+def html_safe(klass):
+    """
+    A decorator that defines the __html__ method.
+    This helps non django templates to detect classes whose __str__ methods
+    return safe text.
+    """
+    if '__html__' in klass.__dict__:
+        raise ValueError("can't apply @html_safe to %s because it defines "
+                         "__html__()." % klass.__name__)
+    if '__str__' not in klass.__dict__ and '__unicode__' not in klass.__dict__:
```

Comment:
If I used hasattr it would fail in this case:
```python
class BaseClass(object):
    def __html__(self):
        # defines __html__ on its own
        return 'some html content'

    def __str__(self):
        return 'some non html content'

@html_safe
class Subclass(BaseClass):
    def __str__(self):
        # overrides __str__ and is marked as html_safe
        return 'some html safe content'
```
Because `hasattr(Subclass, '__html__')` is true but `'__html__' in Subclass.__dict__` is false.
Maybe I should add a test to reflect this behavior.